otx/algorithms/common/adapters/mmcv/hooks/recording_forward_hook.py

```diff
@@ -69,10 +70,25 @@ def _recording_forward(
         self, _: torch.nn.Module, x: torch.Tensor, output: torch.Tensor
     ):  # pylint: disable=unused-argument
         tensors = self.func(output)
-        tensors = tensors.detach().cpu().numpy()
-        for tensor in tensors:
+        if isinstance(tensors, torch.Tensor):
+            tensors_np = tensors.detach().cpu().numpy()
+        elif isinstance(tensors, np.ndarray):
+            tensors_np = tensors
+        else:
+            self._torch_to_numpy_from_list(tensors)
+            tensors_np = tensors
+
+        for tensor in tensors_np:
             self._records.append(tensor)
 
+    def _torch_to_numpy_from_list(self, tensor_list: List[Optional[torch.Tensor]]):
+        if isinstance(tensor_list[0], list):
```

Comment:
Assuming that this is a recursive function that can handle nested lists, `isinstance(tensor_list[0], list)` defines a recursive case when the elements in the list are still lists rather than tensors -> then we go to the next level down.

Reply:
Thanks for your reply! As my understand, this code transforms only first element from list to numpy and leaves others be. I wonder it's your intention.

Reply:
Actually this function convert all tensors to numpy from the nested list. The first element is just used to check if we reached the lower-level list.

Reply:
I may think wrongly, but I think it doesn't work when second element is also list. Isn't code need to be changed as below to transform all list?
```
    def _torch_to_numpy_from_list(self, tensor_list: List[Optional[torch.Tensor]]):
        for i in range(len(tensor_list)):
            if isinstance(tensor_list[i], list):
                self._torch_to_numpy_from_list(tensor_list[i])
            elif isinstance(tensor_list[i], torch.Tensor):
                tensor_list[i] = tensor_list[i].detach().cpu().numpy()
```
It could be wrong, so please let me know when it is :)

Reply:
Your proposal is more general, handles case when the list can contain tensor and another list, while current approach is limited, although still working. I adopt your proposal. Thanks!